packages/flutter/lib/src/material/text_button.dart

```diff
@@ -200,9 +200,9 @@ class TextButton extends ButtonStyleButton {
       : disabledIconColor == null
         ? ButtonStyleButton.allOrNull<Color?>(iconColor)
         : _TextButtonDefaultIconColor(iconColor, disabledIconColor);
-    final MaterialStateProperty<MouseCursor>? mouseCursor = (enabledMouseCursor == null && disabledMouseCursor == null)
+    final MaterialStateProperty<MouseCursor?>? mouseCursor = (enabledMouseCursor == null && disabledMouseCursor == null)
```

Comment:
This fix looks fine however maybe it would be sufficient to just have
```dart
final MaterialStateProperty<MouseCursor?> = _TextButtonDefaultMouseCursor(enabledMouseCursor, disabledMouseCursor);
```
This extra special case doesn't seem to be worth the trouble.

Reply:
Right! I totally missed that. Thank you

Reply:
For the sake of consistency, I've also this line other material buttons. For bother buttons it's just a code restructure.